src/main/java/seedu/address/logic/commands/ImportVolunteerCsvCommand.java

```diff
@@ -0,0 +1,128 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ADDRESS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_BIRTHDAY;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_GENDER;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import seedu.address.logic.CommandHistory;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.parser.ArgumentMultimap;
+import seedu.address.logic.parser.ArgumentTokenizer;
+import seedu.address.logic.parser.ParserUtil;
+import seedu.address.logic.parser.Prefix;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.Model;
+import seedu.address.model.tag.Tag;
+import seedu.address.model.volunteer.Address;
+import seedu.address.model.volunteer.Birthday;
+import seedu.address.model.volunteer.Email;
+import seedu.address.model.volunteer.Gender;
+import seedu.address.model.volunteer.Name;
+import seedu.address.model.volunteer.Phone;
+import seedu.address.model.volunteer.Volunteer;
+
+/**
+ * Imports a list of volunteers in csv format
+ */
+public class ImportVolunteerCsvCommand extends Command {
+    public static final String FILE_ERROR = "Unable to locate the file directory given.\n"
+            + "Please check your file format and try again";
+
+    public static final String COMMAND_WORD = "importvolunteercsv";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Imports a CSV file of the volunteer "
+            + "given the directory to the file.\n"
+            + "The CSV must have the following format: \n"
+            + " Name, Phone, Address, Email, Birthday, Gender, Tags, VolunteerID";
```

Comment:
```suggestion
            + "Name, Phone, Address, Email, Birthday, Gender, Tags, VolunteerID";
```
Not sure if the extra space here is necessary - do correct me if I'm wrong